### PR TITLE
Change load_cube to allowed None to be returned if no filepath.

### PR DESCRIPTION
**Pull request by neilCrosswaite**
The CLI's often need to optionally load cubes. This means that the cubes can be either Nonetype or a Cube. 
This move the responsibility of that to a bool kwarg rather than extra if statements throughout the CLI's.

This has taken a single check at the start of the load function and a refactor across 10 CLI's.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/tests/utilities/test_load.py [hunk 1]
```diff
@@ -220,6 +220,15 @@ def test_lazy_load(self):
...
         self.assertTrue(result.has_lazy_data())
 
+    def test_none_file_with_return_none(self):
+        """Test that return none works with None as filepath."""
+        self.assertIsNone(load_cube(None, return_none=True))
+
+    def test_none_file_without_return_none(self):
+        """Test that without return_none it raises an with None as filepath."""
+        with self.assertRaises(TypeError):
+            load_cube(None)
```
Comment (LaurenceBeard) [inline lib/improver/tests/utilities/test_load.py]: if you insist ;)

File: lib/improver/utilities/load.py [hunk 2]
```diff
@@ -50,16 +51,22 @@ def load_cube(filepath, constraints=None, no_lazy_load=False):
             This can be in the form of an iris.Constraint or could be a string
             that is intended to match the name of the cube.
             The default is None.
-        no_lazy_load (bool)
+        no_lazy_load (bool):
             If True, bypass cube deferred (lazy) loading and load the whole
             cube into memory. This can increase performance at the cost of
             memory. If False (default) then lazy load.
+        return_none (bool):
```
Comment (LaurenceBeard) [inline lib/improver/utilities/load.py]: allow_none may be clearer?

**Comment by LaurenceBeard** [comment]
I actually prefer wrapping/decorating the function as it doesn't pollute the normal load_cube. 
But I imagine this will be rethought when we standardise the loading/saving.
> allow_none(load_cube(cube))

**Comment by LaurenceBeard** [comment]
this could be extended to the general case for:
`allow_none(load_json(json))`

File: lib/improver/tests/utilities/test_load.py [hunk 3]
```diff
@@ -220,6 +220,15 @@ def test_lazy_load(self):
         result = load_cube(self.filepath)
         self.assertTrue(result.has_lazy_data())
 
+    def test_none_file_with_allow_none(self):
+        """Test that return none works with None as filepath."""
+        self.assertIsNone(load_cube(None, allow_none=True))
+
+    def test_none_file_without_allow_none(self):
+        """Test that without allow_none it raises an with None as filepath."""
```
[the remark this answers is not in the document]
Comment (neilCrosswaite) [reply lib/improver/tests/utilities/test_load.py]: Done

**Pull request merged**